tests/test_constants.py

```diff
@@ -7,12 +8,390 @@
+               resizeMode=RESIZE_MODE_UNRESIZEABLE
+               mLastReportedMultiWindowMode=false mLastReportedPictureInPictureMode=false
+              mStartActivity:
+               packageName=com.sony.dtv.tvx processName=com.sony.dtv.tvx
+               launchedFromUid=10134 launchedFromPackage=com.sony.dtv.tvx userId=0
+               app=null
+               Intent { act=android.intent.action.VIEW dat=content://android.media.tv/passthrough/com.sony.dtv.tvinput.external%2F.ExternalTvInputService%2FHDMI100004 flg=0x10400000 cmp=com.sony.dtv.tvx/.MainActivity (has extras) }
+               frontOfTask=false task=TaskRecord{9eecc8c #2830 A=com.sony.dtv.tvx U=0 StackId=1 sz=1}
+               taskAffinity=com.sony.dtv.tvx
+               realActivity=com.sony.dtv.tvx/.MainActivity
+               baseDir=/system/priv-app/Tvx/Tvx.apk
+               dataDir=/data/user/0/com.sony.dtv.tvx
+               stateNotNeeded=false componentSpecified=false mActivityType=0
+               compat=null labelRes=0x7f090048 icon=0x7f02006d theme=0x7f0b01c2
+               mLastReportedConfigurations:
+                mGlobalConfig={1.0 ?mcc?mnc [en_GB] ldltr sw540dp w960dp h540dp 320dpi lrg long hdr land television -touch -keyb/v/h dpad/v appBounds=Rect(0, 0 - 1920, 1080) s.3}
+                mOverrideConfig={0.0 ?mcc?mnc ?localeList ?layoutDir ?swdp ?wdp ?hdp ?density ?lsize ?long ?ldr ?wideColorGamut ?orien ?uimode ?night ?touch ?keyb/?/? ?nav/?}
+               CurrentConfiguration={1.0 ?mcc?mnc [en_GB] ldltr sw540dp w960dp h540dp 320dpi lrg long hdr land television -touch -keyb/v/h dpad/v appBounds=Rect(0, 0 - 1920, 1080) s.3}
+               launchFailed=false launchCount=0 lastLaunchTime=0
+               haveState=true icicle=null
+               state=INITIALIZING stopped=false delayedResume=false finishing=false
+               keysPaused=false inHistory=false visible=false sleeping=false idle=false mStartingWindowState=STARTING_WINDOW_NOT_SHOWN
+               fullscreen=true noDisplay=false immersive=false launchMode=3
+               frozenBeforeDestroy=false forceNewConfig=false
+               mActivityType=APPLICATION_ACTIVITY_TYPE
+               resizeMode=RESIZE_MODE_RESIZEABLE
+               mLastReportedMultiWindowMode=false mLastReportedPictureInPictureMode=false
+               supportsPictureInPicture=true
+               supportsPictureInPictureWhilePausing: false
+              mIntent=Intent { act=android.intent.action.VIEW dat=content://android.media.tv/passthrough/com.sony.dtv.tvinput.external/.ExternalTvInputService/HDMI100004 flg=0x10400000 cmp=com.sony.dtv.tvx/.MainActivity (has extras) }
+              mLaunchSingleTop=false mLaunchSingleInstance=true mLaunchSingleTask=false mLaunchFlags=0x10000000 mDoResume=true mAddingToTask=false
+        """
+        hdmi_input = self._hdmi_input(dumpsys_output)
+
+        self.assertEqual(hdmi_input, '')
```

Comment:
Going forward I would highly recommend to enforce tests cases that actually assert commands against dumpsys output fixtures, which would for sure give me more confidence about (future) changes made to these constants.

Reply:
That's a good idea. I'd thought about it but never took action on it, so thanks for contributing this.